CHANGELOG.md

```diff
@@ -4,6 +4,204 @@ All notable changes to this project will be documented in this file.
+
+### Features
+
+- [`worker`] Upgrade neon to 0.10 ([#675](https://github.com/temporalio/sdk-typescript/pull/675))
+- :boom: Do not allow undefined in PayloadConverter.toPayload ([#672](https://github.com/temporalio/sdk-typescript/pull/672))
+
+  BREAKING CHANGE: `PayloadConverter.toPayload` can no longer return `undefined`.
+
+  NOTE: This change doesn't apply to `PayloadConverterWithEncoding.toPayload` where the function should return `undefined` to mark that the converter doesn't handle a value.
+
+- :boom: Add upsertSearchAttributes and more ([#657](https://github.com/temporalio/sdk-typescript/pull/657))
+
+  - Added and changed `WorkflowInfo` fields
+  - Add `upsertSearchAttributes` Workflow function
+  - Make Search Attributes always arrays
+
+  - Fixes [#314](https://github.com/temporalio/sdk-typescript/pull/314)
+  - Fixes [#445](https://github.com/temporalio/sdk-typescript/pull/445)
+  - Closes [#576](https://github.com/temporalio/sdk-typescript/pull/576)
+  - Closes [#357](https://github.com/temporalio/sdk-typescript/pull/357)
+
+- :boom: Eager and lazy Connection variants + static metadata + worker shutdown refactor ([#682](https://github.com/temporalio/sdk-typescript/pull/682))
+
+- Upgrade to latest sdk-core
+- Fix test-payload-converter not awaiting on promises
+- Simplify Connection metadata API and support static metadata
+- Deprecate `NativeConnection.create` in favor of `connect` method
+- Add lint rule: `@typescript-eslint/no-floating-promises`
+- Close `TestEnvironment.nativeConnection` in `teardown` method
+
+  BREAKING CHANGE:
+
+  - `Connection` constructor is no longer public, and is replaced with `async Connection.connect` and `Connection.lazy` factory methods.
+  - `Connection.service` was renamed `Connection.workflowService`
+  - `WorkflowClient` constructor now accepts a single options param
+
+    BEFORE:
+
+    ```ts
+    const connection = new Connection(...);
+    const client = new WorkflowClient(connection.service, options);
+    ```
+
+    AFTER:
+
+    ```ts
+    const connection = await Connection.create(...);
+    const client = new WorkflowClient({ connection, ...options });
+    ```
+
+  - Added `Connection.close` and made `Connection.client` protected
+
+    NOTE: It is recommended to reuse `Connection` instances as much as possible.
+    When done using a `Connection`, call `close()` to release any resources held by it.
+
+  - `LOCAL_DOCKER_TARGET` constant was renamed `LOCAL_TARGET`
+
+  - Metrics are now emitted with the `temporal_` prefix by default to be consistent with other SDKs, in the near future this can be disabled by setting `TelemetryOptions.noTemporalPrefixForMetrics` to `true`.
+
+  Closes [#607](https://github.com/temporalio/sdk-typescript/pull/607)
+  Closes [#677](https://github.com/temporalio/sdk-typescript/pull/677)
+  Closes [#452](https://github.com/temporalio/sdk-typescript/pull/452)
```

Comment:
```suggestion
  - Metrics are now emitted with the `temporal_` prefix by default to be consistent with other SDKs, in the near future this can be disabled by setting `TelemetryOptions.noTemporalPrefixForMetrics` to `true`.

  - Closes [#607](https://github.com/temporalio/sdk-typescript/pull/607)
  - Closes [#677](https://github.com/temporalio/sdk-typescript/pull/677)
  - Closes [#452](https://github.com/temporalio/sdk-typescript/pull/452)
```